src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs

```diff
@@ -279,7 +276,7 @@ internal int SerializeLogRecord(LogRecord logRecord)
                 bodyPopulated = true;
                 continue;
             }
-            else if (this.m_customFields == null || this.m_customFields.ContainsKey(entry.Key))
+            else if (this.m_customFields == null || entry.Key == "name" || this.m_customFields.ContainsKey(entry.Key))
```

Comment:
Would this break existing users who don't specify customFields AND who use `name` in their key-value pairs?

Consider adding positive/negative test cases.

Reply:
```csharp
exporterOptions.CustomFields = new string[] { "food", "price" };
logger.LogInformation(
    "Hello from {name} {price} {color} {weight}.",
    "apple", 3.99, "red", 1.99);
```

Reply:
You are right, I changed to not override Part B "name" when using customFields without "name" being included in the list